packages/server/src/middleware/rate-limit.js

```diff
@@ -0,0 +1,163 @@
+// Based on https://github.com/nfriedly/express-rate-limit (MIT)
+
+const defaults = {
+  // How long to keep records of requests in memory (milliseconds)
+  windowMs: 1000,
+  // Max number of recent connections during `window` milliseconds before sending a 429 response
+  max: 15,
+  // 429 status = Too Many Requests (RFC 6585)
+  statusCode: 429,
+  // Send custom rate limit header with limit and remaining
+  headers: true,
+  // Do not count failed requests (status >= 400)
+  skipFailedRequests: false,
+  // Do not count successful requests (status < 400)
+  skipSuccessfulRequests: false,
+  // Allows to create custom keys (by default user IP is used)
+  keyGenerator: req => req.headers['x-forwarded-for'] || req.connection.remoteAddress || req.socket.remoteAddress || (req.connection.socket ? req.connection.socket.remoteAddress : null),
+  // Skip certain requests
+  skip: () => false,
+  // Handler in case of reate limits
+  handler: undefined,
+  // A custom callback when rate limit reached
+  onLimitReached: () => {}
+}
+
+export default function RateLimit(_options) {
+  const options = {
+    ...defaults,
+    ..._options
+  }
+
+  if (typeof options.handler === 'undefined') {
+    options.handler = (req, res) => {
+      const secondsLeft = Math.ceil((+req.rateLimit.resetTime - Date.now()) / 1000)
```

Comment:
How about using hrtime ?

Reply:
Never heard of it, why not :)

Also, we could support [ms](https://www.npmjs.com/package/ms) package for `windowMs`

Reply:
Actually we already have the timer in utils, maybe we can leverage it.

Reply:
hrTime is the legacy version of bigInt. We don't need that amount of accuracy here.